### PR TITLE
Add methanol techs

### PR DESCRIPTION
**Pull request by p-glaum**
Closes # (if applicable).

Add and extend methanol technologies from Fabain's import paper.

## Checklist

- [x] I tested my contribution locally and it seems to work fine.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in all of `config.default.yaml`.
- [x] Changes in configuration options are also documented in `doc/configtables/*.csv`.
- [x] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

File: config/config.default.yaml [hunk 1]
```diff
@@ -591,6 +590,19 @@ sector:
     # - onshore  # more than 50 km from sea
   - nearshore    # within 50 km of sea
     # - offshore
+  methanol: false
```
Comment (lisazeyen) [inline config/config.default.yaml:593]: it is maybe better to have all of the technologies as a nested dictionary, if you want to have this switch
```
methanol:
    enable: true
    regional_methanol_demand: false
    [...]
```

File: scripts/prepare_sector_network.py [hunk 2]
```diff
@@ -778,6 +778,331 @@ def add_allam(n, costs):
...
+            efficiency2=costs.at["methanolisation", "carbondioxide-input"],
+            lifetime=25,
+        )
+
+    if types["ccgt_cc"]:
+        logger.info(
+            "Adding methanol CCGT power plants with post-combustion carbon capture."
+        )
+
+        # TODO consider efficiency changes / energy inputs for CC
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:896]: should this be done before the merge?

File: scripts/prepare_sector_network.py [hunk 3]
```diff
@@ -778,6 +778,331 @@ def add_allam(n, costs):
...
+
+        # efficiency * EUR/MW * (annuity + FOM)
+        capital_cost = 0.58 * 916e3 * (calculate_annuity(25, 0.07) + 0.035)
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:899]: better to add the numbers to the technology data

File: scripts/prepare_sector_network.py [hunk 4]
```diff
@@ -778,6 +778,331 @@ def add_allam(n, costs):
...
+        n.madd(
+            "Link",
+            nodes,
+            suffix=" OCGT methanol",
+            bus0=spatial.methanol.nodes,
+            bus1=nodes,
+            bus2="co2 atmosphere",
+            carrier="OCGT methanol",
+            p_nom_extendable=True,
+            capital_cost=0.35
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:939]: same for the costs here

File: scripts/prepare_sector_network.py [hunk 5]
```diff
@@ -2254,6 +2579,47 @@ def add_heat(n: pypsa.Network, costs: pd.DataFrame, cop: xr.DataArray):
                 )
 
 
+def add_methanol(n, costs):
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:2582]: can one adapt the function `def add_carrier_buses`[here](https://github.com/PyPSA/pypsa-eur/blob/05dba2d36b5a9b640bbb659d627f21f7ae15b5a1/scripts/prepare_sector_network.py#L523) to work with methanol as well?

**Review by lisazeyen — CHANGES_REQUESTED** [review]

File: config/config.default.yaml (hunk 1 above)
Comment (p-glaum) [reply config/config.default.yaml:593]: changes it to your suggestions.

File: scripts/prepare_sector_network.py (hunk 5 above)
Comment (p-glaum) [reply scripts/prepare_sector_network.py:2582]: I previously thought the same but I decided against it. However, now I do not see any reason to not use ``add_carrier_busses`` so I used it again.

File: scripts/prepare_sector_network.py (hunk 3 above)
Comment (p-glaum) [reply scripts/prepare_sector_network.py:899]: used similar to h2 turbine CCGT and OCGT cost assumptions

File: scripts/prepare_sector_network.py (hunk 4 above)
Comment (p-glaum) [reply scripts/prepare_sector_network.py:939]: same as for CCGT

**Review by lisazeyen — APPROVED** [review]

File: scripts/prepare_sector_network.py [hunk 6]
```diff
@@ -777,6 +777,312 @@ def add_allam(n, costs):
...
+            "Link",
+            nodes,
+            suffix=" allam methanol",
+            bus0=spatial.methanol.nodes,
+            bus1=nodes,
+            bus2=spatial.co2.df.loc[nodes, "nodes"].values,
+            bus3="co2 atmosphere",
+            carrier="allam methanol",
+            p_nom_extendable=True,
+            capital_cost=0.59
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:848]: Can you add the numbers here to the technology data instead? @Irieo might have some sources

File: scripts/prepare_sector_network.py [hunk 7]
```diff
@@ -777,6 +777,312 @@ def add_allam(n, costs):
...
+            nodes,
+            suffix=" CCGT methanol CC",
+            bus0=spatial.methanol.nodes,
+            bus1=nodes,
+            bus2=spatial.co2.df.loc[nodes, "nodes"].values,
+            bus3="co2 atmosphere",
+            carrier="CCGT methanol CC",
+            p_nom_extendable=True,
+            capital_cost=capital_cost_cc,
+            marginal_cost=2,
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:907]: same for the marginal cost here, would be great to refer to the technology data and or add as source

File: scripts/prepare_sector_network.py [hunk 8]
```diff
@@ -777,6 +777,312 @@ def add_allam(n, costs):
...
+            "Link",
+            nodes,
+            suffix=" OCGT methanol",
+            bus0=spatial.methanol.nodes,
+            bus1=nodes,
+            bus2="co2 atmosphere",
+            carrier="OCGT methanol",
+            p_nom_extendable=True,
+            capital_cost=costs.at["OCGT", "fixed"] * costs.at["OCGT", "efficiency"],
+            marginal_cost=2,
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:929]: same as above

File: scripts/prepare_sector_network.py [hunk 9]
```diff
@@ -2998,25 +3330,26 @@ def add_industry(n, costs):
...
-    )
+        n.madd(
+            "Store",
+            spatial.methanol.nodes,
+            suffix=" Store",
+            bus=spatial.methanol.nodes,
+            e_nom_extendable=True,
+            e_cyclic=True,
+            carrier="methanol",
+            capital_cost=0.02,
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:3351]: same here

File: scripts/prepare_sector_network.py (hunk 8 above)
Comment (p-glaum) [reply scripts/prepare_sector_network.py:929]: done

File: scripts/prepare_sector_network.py (hunk 6 above)
Comment (p-glaum) [reply scripts/prepare_sector_network.py:848]: I added some dummy values which can be update later

File: scripts/prepare_sector_network.py [hunk 10]
```diff
@@ -819,11 +819,315 @@ def add_allam(n, costs):
...
+            "Link",
+            nodes,
+            suffix=" CCGT methanol",
+            bus0=spatial.methanol.nodes,
+            bus1=nodes,
+            bus2="co2 atmosphere",
+            carrier="CCGT methanol",
+            p_nom_extendable=True,
+            capital_cost=capital_cost,
+            marginal_cost=2,
```
Comment (lisazeyen) [inline scripts/prepare_sector_network.py:922]: should these marginal costs be in the technology-data ?

**Review by lisazeyen — CHANGES_REQUESTED** [review]

File: scripts/prepare_sector_network.py (hunk 10 above)
Comment (p-glaum) [reply scripts/prepare_sector_network.py:922]: oh forgot them. Now they are there

File: scripts/prepare_sector_network.py [hunk 11]
```diff
@@ -819,11 +819,315 @@ def add_allam(n, costs):
...
+            capital_cost=costs.at["OCGT", "fixed"] * costs.at["OCGT", "efficiency"],
+            marginal_cost=costs.at["OCGT", "VOM"] * costs.at["OCGT", "efficiency"],
+            efficiency=costs.at["OCGT", "efficiency"],
+            efficiency2=costs.at["methanolisation", "carbondioxide-input"],
+            lifetime=costs.at["OCGT", "lifetime"],
+        )
+
+
+def add_methanol_to_olefins(n, costs):
+    nodes = pop_layout.index
```
[the remark this answers is not in the document]
Comment (p-glaum) [reply scripts/prepare_sector_network.py:985]: done

File: scripts/prepare_sector_network.py [hunk 12]
```diff
@@ -2304,6 +2608,32 @@ def add_heat(n: pypsa.Network, costs: pd.DataFrame, cop: xr.DataArray):
...
+def add_methanol(n, costs):
+
+    methanol_options = options["methanol"]
+    if not any(methanol_options.values()):
+        return
+
+    logger.info("Add methanol")
+    add_carrier_buses(n, "methanol")
+
+    if n.buses.carrier.str.contains("biomass").any():
```
[the remark this answers is not in the document]
Comment (p-glaum) [reply scripts/prepare_sector_network.py:2620]: done

File: scripts/prepare_sector_network.py (hunk 2 above)
Comment (p-glaum) [reply scripts/prepare_sector_network.py:933]: I think this is not considered in other CC techs so far, but at one points we could do it for all CC techs.

**Review by lisazeyen — APPROVED** [review]

**Pull request merged**